web/src/server/api/routers/models.ts

```diff
@@ -132,19 +132,42 @@ export const modelRouter = createTRPCRouter({
         });
       }
 
-      const createdModel = await ctx.prisma.model.create({
-        data: {
-          projectId: input.projectId,
-          modelName: input.modelName,
-          matchPattern: input.matchPattern,
-          startDate: input.startDate,
-          inputPrice: input.inputPrice,
-          outputPrice: input.outputPrice,
-          totalPrice: input.totalPrice,
-          unit: input.unit,
-          tokenizerId: input.tokenizerId,
-          tokenizerConfig: input.tokenizerConfig,
-        },
+      const createdModel = await ctx.prisma.$transaction(async (tx) => {
+        const createdModel = await tx.model.create({
+          data: {
+            projectId: input.projectId,
+            modelName: input.modelName,
+            matchPattern: input.matchPattern,
+            startDate: input.startDate,
+            inputPrice: input.inputPrice,
+            outputPrice: input.outputPrice,
+            totalPrice: input.totalPrice,
+            unit: input.unit,
+            tokenizerId: input.tokenizerId,
+            tokenizerConfig: input.tokenizerConfig,
+          },
+        });
+
+        // Populate prices table
+        const prices = [
+          { usageType: "input", price: input.inputPrice },
+          { usageType: "output", price: input.outputPrice },
+          { usageType: "total", price: input.totalPrice },
+        ];
+
+        for (const { usageType, price } of prices) {
+          if (price != null) {
+            await tx.price.create({
+              data: {
+                modelId: createdModel.id,
+                usageType,
+                price,
+              },
+            });
+          }
+        }
```

Comment:
Is there a strong reason to do this one by one? A Promise.all would likely return significantly faster here for a longer list and reduce the time in which the transaction is pending.

Reply:
Thanks, yes this is unoptimized since sequential. Performance diffs should be low however as there are only max a handful of prices per model